src/comm/TCPSocket.cc

```diff
@@ -21,8 +21,10 @@ ENABLE_WARNING(effc++)
 #include "comm/Variables.h"
 
 using namespace comm;
-
-TCPSocket::TCPSocket(int socket_fd) : _socket_fd(socket_fd) {}
+TCPSocket::TCPSocket(int socket_fd, const sockaddr_in& address)
+    : _socket_fd(socket_fd), _source_family(AF_INET), _source(address)
```

Comment:
nit: If `sockaddr_in` is moveable, prefer pass-by-value and move in cases where the method makes a copy:
```suggestion
TCPSocket::TCPSocket(int socket_fd, sockaddr_in address)
    : _socket_fd(socket_fd), _source_family(AF_INET), _source(std::move(address))
```

Reply:
I'm a fan of move, but know it can be tricky. I'll check the conditions.

Reply:
since sockaddr_in is a struct inside a extern "C", it can't have an automatic move constructor generated, so a copy is the best we can ask for, unless I'm reading things wrong.